File: js/factorial.js
var py = require("./py");
console.log ("Testing factorial");
factorial = py.def ({}, function (n) {
    if (n > 0) {
        return n * factorial ([n - 1], {});
    } else {
        return 1;
    };
});
console.log ("Factorial of 2 is", factorial ([2], {}));
console.log ("Factorial of 3 is", factorial ([3], {}));
console.log ("Factorial of 4 is", factorial ([4], {}));
console.log ("Factorial of 5 is", factorial ([5], {}));

File: static/factorial.py
from pythonscript import js
js('var py = require("./py")')
print "Testing factorial"
def factorial(n):
    if (n > 0):
        return n * factorial(n-1)
    else:
        return 1

print "Factorial of 2 is", factorial(2)
print "Factorial of 3 is", factorial(3)
print "Factorial of 4 is", factorial(4)
print "Factorial of 5 is", factorial(5)
